File: packages/jsonresume-theme-professional/src/ui/Volunteer.js
import Section from './Section';
import Experience from './Experience';

const Volunteer = ({ volunteer }) => {
  if (!volunteer) {
    return null;
  }

  return (
    <div>
      <Section title="Volunteer">
        {volunteer.map((w, key) => {
          return (
            <Experience
              title={w.position}
              subTitle={w.organization}
              startDate={w.startDate}
              endDate={w.endDate}
              summary={w.summary}
              highlights={w.highlights}
              key={key}
            />
          );
        })}
      </Section>
    </div>
  );
};

export default Volunteer;

File: packages/core-rust/src/resume/volunteer.rs
use crate::validation::{validate_date, validate_url};
use serde::{Deserialize, Serialize};
use serde_valid::Validate;

#[derive(Debug, Serialize, Deserialize, Validate)]
pub struct Volunteer {
    pub organization: Option<String>,
    pub position: Option<String>,
    #[validate(custom = validate_url)]
    pub url: Option<String>,
    #[serde(rename = "startDate")]
    #[validate(custom = validate_date)]
    pub start_date: Option<String>,
    #[serde(rename = "endDate")]
    #[validate(custom = validate_date)]
    pub end_date: Option<String>,
    pub summary: Option<String>,
    pub highlights: Vec<String>,
}
